tools/mobdbconverter.py

```diff
@@ -0,0 +1,269 @@
+                            continue
+                        value = int(value)
+                        if value not in itemDb:
+                            w.write(w, "// Error: mvp drop with id {0} not found in item_db.conf\n".format(value))
+                        else:
+                            printField2(w, itemDb[value], chance)
+                    endGroup(w)
+                if isHaveData(fields, 37, 10):
+                    startGroup(w, "Drops")
+                    for f in range(0, 10):
+                        value = fields[37 + f * 2]
+                        chance = fields[38 + f * 2]
+                        if value == "" or value == "0" or chance == "" or chance == "0":
+                            continue
+                        value = int(value)
+                        if value not in itemDb:
+                            w.write(w, "// Error: drop with id {0} not found in item_db.conf\n".format(value))
+                        else:
+                            printField2(w, itemDb[value], chance)
+                    endGroup(w)
+                w.write("},\n")
+            printFooter(w)
+
+def readItemDB(inFile, itemDb):
+    itemId = 0
+    itemName = ""
+    started = False
+    with open(inFile, "r") as r:
+        for line in r:
+            line = line.strip()
+            if started == True:
+                if line == "},":
+                    started = False
+                elif line[:10] == "AegisName:":
+                    itemName = line[12:-1]
+                elif line[:3] == "Id:":
+                    try:
+                        itemId = int(line[4:])
+                    except:
+                        started = False
+                if itemId != 0 and itemName != "":
+                    itemName = itemName.replace(".", "")
+                    if itemName[0] >= "0" and itemName[0] <= "9":
+                        itemName = "Num" + itemName
+                    itemDb[itemId] = itemName
+                    started = False
+            else:
+                if line == "{":
+                    started = True
+                    itemId = 0
+                    itemName = ""
+    return itemDb
+
+if len(sys.argv) != 2:
+    printHelp();
+    exit(1)
+startPath = "../"
+itemDb = dict()
+if sys.argv[1] == "re":
+    itemDb = readItemDB(startPath + "db/re/item_db.conf", itemDb)
+    itemDb = readItemDB(startPath + "db/item_db2.conf", itemDb)
+    convertFile(startPath + "db/re/mob_db.txt", startPath + "db/re/mob_db.conf", itemDb)
+    convertFile(startPath + "db/mob_db2.txt", startPath + "db/mob_db2.conf", itemDb)
+elif sys.argv[1] == "pre-re":
+    itemDb = readItemDB(startPath + "db/pre-re/item_db.conf", itemDb)
+    itemDb = readItemDB(startPath + "db/item_db2.conf", itemDb)
+    convertFile(startPath + "db/pre-re/mob_db.txt", startPath + "db/pre-re/mob_db.conf", itemDb)
+    convertFile(startPath + "db/mob_db2.txt", startPath + "db/mob_db2.conf", itemDb)
+else:
+    printHelp();
```

Comment:
it'd be nice if this worked like all other converters out there (parses stdin, outputs to stdout), so that it can be used with arbitrary files (or as a CGI script)